pkg/kubelet/pod_workers.go

```diff
@@ -387,18 +391,20 @@ func newPodWorkers(
 	podCache kubecontainer.Cache,
 ) PodWorkers {
 	return &podWorkers{
-		podSyncStatuses:               map[types.UID]*podSyncStatus{},
-		podUpdates:                    map[types.UID]chan podWork{},
-		lastUndeliveredWorkUpdate:     map[types.UID]podWork{},
-		terminatingStaticPodFullnames: map[string]struct{}{},
-		syncPodFn:                     syncPodFn,
-		syncTerminatingPodFn:          syncTerminatingPodFn,
-		syncTerminatedPodFn:           syncTerminatedPodFn,
-		recorder:                      recorder,
-		workQueue:                     workQueue,
-		resyncInterval:                resyncInterval,
-		backOffPeriod:                 backOffPeriod,
-		podCache:                      podCache,
+		podSyncStatuses:                    map[types.UID]*podSyncStatus{},
+		podUpdates:                         map[types.UID]chan podWork{},
+		lastUndeliveredWorkUpdate:          map[types.UID]podWork{},
+		terminatingStaticPodFullnames:      map[string]struct{}{},
+		startedPodUIDsByFullname:           map[string]types.UID{},
```

Comment:
You only need startedPodUIDsByFullname and can remove terminatingStaticPodFullnames.  IsPodForMirrorPodTerminatingByFullName now must calculate by checking startedPodUIDsByFullname, then checking the podSyncStatus for that UID and see whether it is both terminationRequested and ! terminated.